test/e2e/egress_test.go

```diff
@@ -0,0 +1,209 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/vmware-tanzu/antrea/pkg/apis/crd/v1alpha2"
+)
+
+func TestEgress(t *testing.T) {
+	skipIfNotIPv4Cluster(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	cc := []configChange{
+		{"Egress", "true", true},
+	}
+	ac := []configChange{
+		{"Egress", "true", true},
+	}
+	if err := data.mutateAntreaConfigMap(cc, ac, true, true); err != nil {
+		t.Fatalf("Failed to enable NetworkPolicyStats feature: %v", err)
+	}
+
+	egressNode := controlPlaneNodeName()
+	egressNodeIP := controlPlaneNodeIP()
+	localIP0 := "1.1.1.10"
+	localIP1 := "1.1.1.11"
+	serverIP := "1.1.1.20"
+	fakeServer := "fakeserver"
+
+	// Create a http server in another netns to fake an external server connected to the egress Node.
+	cmd := fmt.Sprintf(`ip netns add %[1]s && \
+ip link add dev %[1]s-a type veth peer name %[1]s-b && \
+ip link set dev %[1]s-a netns %[1]s && \
+ip addr add %[3]s/24 dev %[1]s-b && \
+ip addr add %[4]s/24 dev %[1]s-b && \
+ip link set dev %[1]s-b up && \
+ip netns exec %[1]s ip addr add %[2]s/24 dev %[1]s-a && \
+ip netns exec %[1]s ip link set dev %[1]s-a up && \
+ip netns exec %[1]s ip route replace default via %[3]s && \
+ip netns exec %[1]s /agnhost netexec
+`, fakeServer, serverIP, localIP0, localIP1)
+	if err := data.createPodOnNode(fakeServer, egressNode, agnhostImage, []string{"sh", "-c", cmd}, nil, nil, nil, true, func(pod *v1.Pod) {
+		privileged := true
+		pod.Spec.Containers[0].SecurityContext = &v1.SecurityContext{Privileged: &privileged}
+	}); err != nil {
+		t.Fatalf("Failed to create server Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, fakeServer)
+	if err := data.podWaitForRunning(defaultTimeout, fakeServer, testNamespace); err != nil {
+		t.Fatalf("Error when waiting for Pod '%s' to be in the Running state", fakeServer)
+	}
+
+	localPod := "localpod"
+	remotePod := "remotepod"
+	if err := data.createBusyboxPodOnNode(localPod, egressNode); err != nil {
+		t.Fatalf("Failed to create local Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, localPod)
+	if err := data.podWaitForRunning(defaultTimeout, localPod, testNamespace); err != nil {
+		t.Fatalf("Error when waiting for Pod '%s' to be in the Running state", localPod)
+	}
+	if err := data.createBusyboxPodOnNode(remotePod, workerNodeName(1)); err != nil {
+		t.Fatalf("Failed to create remote Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, remotePod)
+	if err := data.podWaitForRunning(defaultTimeout, remotePod, testNamespace); err != nil {
+		t.Fatalf("Error when waiting for Pod '%s' to be in the Running state", remotePod)
+	}
+
+	getClientIP := func(pod string) (string, string, error) {
```

Comment:
I feel you should add some comments to these funcs.

Reply:
done